src/App.php

```diff
@@ -1247,8 +1248,13 @@ public function runFrontend()
 				$this->module = "login";
 			}
 
+			$router = new App\Router();
```

Comment:
What about loading the router outside of `App` and add it (temporary) to the constructor (until we introduced the whole DI framework).

This will lower the coupling.

Reply:
What about the `DependencyFactory::setUp` method instead?

The main problem with coupling is that we need to call `Hook::callAll` on the `RouteCollection` object that is inside the `Router`. I don't know how to do this gracefully.

Reply:
I'd like to call it like
```php
class Router 
{
  public function callHook()
  {
    Hook::callAll();
  }
}
```

which are used after the App initialization.

It's one part of something I call "dependency circle":
`App` -> calls `Router` -> calls `Hook` -> calls `App`
What if we move the Router call to a position where not "everything" of the current App call is set for a successful "Hook" call. How can a maintainer of the Router class know it?

I'm at work, so I cannot investigate what Hook really needs for the callAll method :-( .. But I would prefer to create the Router outside of the App and inside the App, we call the Hook explicit.

So if we ever want to change the Router behaviour, it is clear that we don't have to think about the App, but it is nevertheless bound at the right place.

Reply:
OK, on a long run, I think it should be:
- `App` initializiation
- `Router` initialization
- `App`->runFrontend(Router $router);

What about that? :-)

Reply:
I now realize we can just call `Hook::callAll($this->router->getRouterCollector())` just before the dispatch.

Thanks for the hints!